WhatsApp/app/src/main/java/com/example/whatsapp/ui/fragments/home/camera/CameraFragmentVM.kt

```diff
@@ -0,0 +1,17 @@
+package com.example.whatsapp.ui.fragments.home.camera
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.example.whatsapp.base.BaseViewModel
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+
+@HiltViewModel
+class CameraFragmentVM @Inject constructor() : BaseViewModel() {
+    private val _isPermissionGranted = MutableLiveData(false)
```

Comment:
I believe we should create a seperate VM for permissions so that we can inject it into other fragments

Reply:
Are there any fragments that will use camera? I can create a permission manager in that case

Reply:
I was referring more to the permission live data, I think we could have a seperate viewmodel for the permissions that we inject in places where we handle permissions. Not entirely sure

Reply:
If @Josh-Owen thinks its fine, he can approve and we get this merged in :)